src/pid.cpp

```diff
@@ -318,6 +323,25 @@ double Pid::computeCommand(double error, double error_dot, ros::Duration dt)
   // Compute the command
   cmd_ = p_term + i_term + d_term;
 
+  // Publish controller state if configured
+  if (publish_state_ && state_publisher_.trylock())
+  {
+    state_publisher_.msg_.header.stamp = ros::Time::now();
```

Comment:
Hmmm, I realize just now that under certain RTOSes, like Xenomai 2.6.X, `ros::Time::now` is a no-no, as there's a Linux system call involved.

Since state publishing is optional and comes disabled by default, I think your proposal is an acceptable compromise.

Could you please document the `publish_state` parameter [here](https://github.com/ros-controls/control_toolbox/blob/indigo-devel/include/control_toolbox/pid.h#L81), and mention that if enabled, it may break real-time guarantees on certain RTOS?. I think that with this, the PR can be merged.